src/utilities.jl

```diff
@@ -14,7 +14,11 @@ macro sync(code)
 end
 
 function versioninfo(io::IO=stdout)
-    println(io, "macOS $(macos_version()), Darwin $(darwin_version())")
+    print(io, "macOS $(macos_version()), Darwin $(darwin_version())")
+    if process_translated()
+        print(io, ", using Rosetta")
+    end
+    println(io)
```

Comment:
What if you implemented the same mechanism forunsupported platforms as in [CUDA.jl](
https://github.com/JuliaGPU/CUDA.jl/blob/2a3cec926ba48f5bb4efbec5c9242259370963fa/src/initialization.jl#L9-L29)
```suggestion
    @assert functional(true)
    
    println(io, "macOS $(macos_version()), Darwin $(darwin_version())")
```

Reply:
isn't `functional()` on its way out?

```
    # TODO: make errors here (and in submodules/subpackages like cuBLAS and cuDNN) fatal,
    #       and remove functional(), once people sufficiently use weak dependencies.

```

Reply:
In that case this looks good to me.